File: redome/src/main/java/br/org/cancer/modred/model/AvaliacaoNovaBusca.java
package br.org.cancer.modred.model;

import java.io.Serializable;
import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonView;

import br.org.cancer.modred.controller.view.AvaliacaoNovaBuscaView;
import br.org.cancer.modred.model.domain.StatusAvaliacaoNovaBusca;
import br.org.cancer.modred.model.security.Usuario;
import br.org.cancer.modred.service.impl.custom.EntityObservable;
import br.org.cancer.modred.service.impl.observers.AvaliacaoNovaBuscaAprovadaObserver;
import br.org.cancer.modred.service.impl.observers.AvaliacaoNovaBuscaReprovadaObserver;
import br.org.cancer.modred.service.impl.observers.AvaliacaoNovaBuscaSolicitadaObserver;

/**
 * Avaliação da nova busca criada para o paciente.
 * Esta é avaliada pela equipe Redome.
 * 
 * @author Pizão
 *
 */
@Entity
@SequenceGenerator(name = "SQ_AVNB_ID", sequenceName = "SQ_AVNB_ID", allocationSize = 1)
@Table(name = "AVALIACAO_NOVA_BUSCA")
public class AvaliacaoNovaBusca extends EntityObservable implements Serializable {
	private static final long serialVersionUID = -3164067807867300311L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SQ_AVNB_ID")
	@Column(name = "AVNB_ID")
	@JsonView(AvaliacaoNovaBuscaView.ListarTarefas.class)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "PACI_NR_RMR")
	@NotNull
	@JsonView(AvaliacaoNovaBuscaView.ListarTarefas.class)
	private Paciente paciente;
	
	@Column(name = "AVNB_DT_CRIACAO")
	@JsonView(AvaliacaoNovaBuscaView.ListarTarefas.class)
	@NotNull
	private LocalDateTime dataCriacao;

	@Enumerated(EnumType.STRING)
	@Column(name = "AVNB_TX_STATUS")
	@NotNull
	@JsonView(AvaliacaoNovaBuscaView.ListarTarefas.class)
	private StatusAvaliacaoNovaBusca status;

	@ManyToOne
	@JoinColumn(name = "USUA_ID_AVALIADOR")
	private Usuario avaliador;
	
	@Column(name = "AVNB_DT_AVALIADO")
	private LocalDateTime dataAvaliado;

	@Column(name = "AVNB_TX_JUSTIFICATIVA")
	private String justificativa;

	/**
	 * Construtor que define os observers envolvidos no
	 * processamento desta entidade.
	 */
	public AvaliacaoNovaBusca() {
		super();
		addObserver(AvaliacaoNovaBuscaSolicitadaObserver.class);
		addObserver(AvaliacaoNovaBuscaAprovadaObserver.class);
		addObserver(AvaliacaoNovaBuscaReprovadaObserver.class);
	}


	public Long getId() {
		return id;
	}


	public void setId(Long id) {
		this.id = id;
	}


	public LocalDateTime getDataAvaliado() {
		return dataAvaliado;
	}


	public void setDataAvaliado(LocalDateTime dataAvaliado) {
		this.dataAvaliado = dataAvaliado;
	}


	public Paciente getPaciente() {
		return paciente;
	}


	public void setPaciente(Paciente paciente) {
		this.paciente = paciente;
	}


	public StatusAvaliacaoNovaBusca getStatus() {
		return status;
	}


	public void setStatus(StatusAvaliacaoNovaBusca status) {
		this.status = status;
	}


	public Usuario getAvaliador() {
		return avaliador;
	}


	public void setAvaliador(Usuario avaliador) {
		this.avaliador = avaliador;
	}


	public String getJustificativa() {
		return justificativa;
	}


	public void setJustificativa(String justificativa) {
		this.justificativa = justificativa;
	}


	public LocalDateTime getDataCriacao() {
		return dataCriacao;
	}


	public void setDataCriacao(LocalDateTime dataCriacao) {
		this.dataCriacao = dataCriacao;
	}
	
}

File: redome-front/src/app/shared/classes/avaliacao.nova.busca.ts
import { BaseEntidade } from "app/shared/base.entidade";
import { Paciente } from "../../paciente/paciente";
import { StatusAvaliacaoNovaBusca } from "../enums/status.avaliacao.nova.busca";
import { UsuarioLogado } from "../dominio/usuario.logado";

/**
 * Avaliação da solicitação para início de novo processo de busca
 * para o paciente que já possuia cadastro no Redome.
 * 
 * @author Pizão
 */
export class AvaliacaoNovaBusca extends BaseEntidade {
    
    private _id: number;
    private _paciente: Paciente;
	private _dataCriacao: Date;
    private _status: StatusAvaliacaoNovaBusca;
    private _avaliador: UsuarioLogado;
    private _justificativa: string;


	public get id(): number {
		return this._id;
	}

	public set id(value: number) {
		this._id = value;
    }
    
	public get paciente(): Paciente {
		return this._paciente;
    }
    
	public set paciente(value: Paciente) {
		this._paciente = value;
	}

	public get dataCriacao(): Date {
		return this._dataCriacao;
	}

	public set dataCriacao(value: Date) {
		this._dataCriacao = value;
	}

	public get status(): StatusAvaliacaoNovaBusca {
		return this._status;
	}

	public set status(value: StatusAvaliacaoNovaBusca) {
		this._status = value;
	}

	public get avaliador(): UsuarioLogado {
		return this._avaliador;
	}

	public set avaliador(value: UsuarioLogado) {
		this._avaliador = value;
	}

	public get justificativa(): string {
		return this._justificativa;
	}

	public set justificativa(value: string) {
		this._justificativa = value;
	}
    

    public jsonToEntity(res: any) {
        throw new Error("Method not implemented.");
    }

}